sql/handler.h

```diff
@@ -1405,6 +1405,21 @@ struct handlerton
    int  (*recover)(handlerton *hton, XID *xid_list, uint len);
    int  (*commit_by_xid)(handlerton *hton, XID *xid);
    int  (*rollback_by_xid)(handlerton *hton, XID *xid);
+   /*
+     recover_rollback_by_xid is optional. If set, it will be called instead
+     of rollback_by_xid when transactions should be rollback at server startup.
+
+     This function should just changed the transaction's state from prepared to
+     active, and rollback the transaction asynchronously by a background
+     thread. Thus rolling back a long transaction will not block server startup.
+   */
+   int  (*recover_rollback_by_xid)(const XID &xid);
```

Comment:
I think the argument should be a pointer (also as elsewhere). After all an implementation's method can not understand C++ references.

Reply:
sure, updated